### ISSUE TITLE
CVE-2014-3539 python-rope: pickle.load of remotely supplied data with no authentication

### ISSUE THREAD

**Issue by nixon** [python-rope/rope#105]
https://bugzilla.redhat.com/show_bug.cgi?id=1116485

**Comment by mcepl** [comment]
I am aware of it, and I was working on it https://gitorious.org/rope/rope/source/CVE-2014-3539 but I have never been able to pull off a good automatized reproducer for test. Any merge requests for that branch would be very very welcome.

**Comment by vpereira** [comment]
Hi, maybe this is already a reproducer?

````
import rope.base.pycore
import rope.base.project

myproject = rope.base.project.Project('.')
res = myproject.get_resource("mod2.py")
myproject.pycore.run_module(res)
````

**Comment by apnewberry** [comment]
Hi, I'm a fan of rope, but am concerned about using it due to this issue. I'm wondering if you might be willing to share your current thinking? Thanks.


In case it's helpful, Debian seems to have a limited patch.
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=777525
https://sources.debian.org/src/rope/0.10.3-1/debian/patches/CVE-2014-3539.patch/

**Comment by soupytwist** [comment]
Hi, I know this has been sitting for far too long. I am working on the fix for this issue.

**Comment by soupytwist** [comment]
I have a proposed fix in #251. I'm not sure who will review it who is very familiar with this codebase (or if there is such a person anymore). This does need to be resolved, so I will get at least another competent pair of eyes on it. Please anyone take a look as well if you are interested.

**Comment by mcepl** [comment]
> who is very familiar with this codebase (or if there is such a person anymore

Given the authorship of last commits to ``rope`` I would say that the most competent person to know this codebase is best found by you in the nearest mirror ;).
